examples/run_parallel_and_cleanup.py

```diff
@@ -0,0 +1,114 @@
+"""Script to run all example files"""
+import os
+import subprocess
+import shutil
+import sys
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+#checks if we are in the examples directory
+if not os.path.basename(os.getcwd()) == "examples":
+    print("Run this script from the examples directory.")
+    sys.exit(1)
+# Parse command-line arguments
+if len(sys.argv) > 2:
+    print("Usage: python run_parallel_and_cleanup.py [max_threads]")
+    sys.exit(1)
+
+# Default to 2 threads if no argument is provided
+max_threads = int(sys.argv[1]) if len(sys.argv) == 2 else 4
```

Comment:
Suggestion: Instead of using the hardcoded value of ```4``` threads, you could let the ```ThreadPoolExecutor``` determine the number of ```max_workers``` automatically by passing ```None```. By default, it uses the formula: ```min(32, (os.cpu_count() or 1) + 4)```.